test/NuGet.Clients.Tests/NuGet.Credentials.Test/SecurePluginCredentialProviderBuilderTests.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using NuGet.Common;
+using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.Plugins;
+using Xunit;
+
+namespace NuGet.Credentials.Test
+{
+    public class SecurePluginCredentialProviderBuilderTests
```

Comment:
You have a problem.

You have different tests classes partying on `PluginManager.Instance` potentially in parallel.  Some tests call [`Reinitialize(...)`](https://github.com/NuGet/NuGet.Client/pull/2126/files#diff-c67d2e9114567b403170c640b29bb7a3R99) and some tests call [`Dispose()`](https://github.com/NuGet/NuGet.Client/pull/2126/files#diff-3eddcf79cd5a584a191f9f68d2d1d61aR453).  This will lead to tests interfering with one another and causing seemingly random failures.

One solution would to make `PluginManager`'s constructor public for testing purposes and eliminate the static stuff (except for `Instance`).  The product will still use `PluginManager.Instance` but tests can new up a new `PluginManager` instance to test in total isolation from other tests.

You may even be able to get rid of `Reinitialize(...)` with this approach.